feat: implement st1-phase-2 session prompts

arrnd opened the pull request:
Phase 2 Session Prompts Feature Development

Session 2.1 - query_location_proofs Tool Implementation:

Add comprehensive location proof querying with filtering capabilities
Implement REST API integration with /api/v0/location-proofs endpoint
Add parameter validation for chain, prover, limit, and offset filters
Include proper async handling and structured error responses
Support pagination and geospatial data processing

Session 2.2 - get_location_proof_by_uid Tool Implementation:

Implement specific attestation retrieval by unique identifier
Add UID format validation (66-character hex with 0x prefix)
Create robust error handling for not-found vs malformed cases
Include complete proof data structure with verification details
Add comprehensive test coverage for positive and negative cases

Session 2.3 - get_astral_config Tool & Error Handling Standardization:

Add Astral API configuration endpoint integration
Implement standardized error response format across all tools
Create consistent success/error structure with detailed context
Add enhanced logging for debugging and monitoring
Include chain and schema information retrieval capabilities

arrnd committed:
feat: implement st1-phase-2 session prompts

SethDocherty commented on tests/test_server.py:
```diff
@@ -28,3 +28,51 @@ async def test_server_info_tool():
+    # Test short UID
+    result = await get_location_proof_by_uid("0x123")
+    assert result["success"] is False
+    assert result["error"] == "validation_error"
```
I don't see any tests here for valid UIDs? We should also test for valid cases to ensure that any modifications to the `get_location_proof_by_uid` method do not impact valid results.

SethDocherty replied:
Added the additional tests

SethDocherty committed:
Adds test case for valid location proof by UID

Adds a test case to verify successful retrieval of a location proof using a valid UID from the AstralAPI backend.

This test ensures that the server correctly handles valid UIDs and returns the expected data, including checks for the existence of data, absence of error messages, and verification of the UID in the response.

SethDocherty commented on astral_mcp_server/server.py:
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
+            "message": error_msg,
+            "details": {
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]
```
Took the suggestion into consideration and created a named constant

SethDocherty commented on astral_mcp_server/server.py:
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
+            "details": {
+                "attempted_uid": uid,
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]
```
Took the suggestion into consideration and created a named constant

SethDocherty committed:
refactor: standardize error text length in API responses

SethDocherty commented on astral_mcp_server/server.py:
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
+            "message": error_msg,
+            "details": {
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]  # Limit error text length
```
Took the suggestion into consideration and created a named constant

SethDocherty committed:
feat: add validation for query limit with defined minimum and maximum values

SethDocherty committed:
Refactors and enhances location proof querying

Improves location proof querying by adding validation,
pagination, and GeoJSON support.

This commit streamlines the query process, providing
more robust and flexible data retrieval. It introduces
validation for input parameters, pagination for handling
large datasets, and the option to include GeoJSON data
for spatial analysis. Additionally, it enhances error
handling and logging for better debugging.

SethDocherty committed:
Adds helpers subpackage

Introduces a new subpackage for helper functions,
grouping utilities for GeoJSON handling, data extraction,
and validation.

SethDocherty committed:
Adds GeoJSON helper utilities

Adds a new module with helper functions for working with GeoJSON data.

This module provides functionalities for:

- Finding Point geometries within complex objects.
- Creating Point geometries from latitude/longitude values.
- Parsing location strings into Point geometries, supporting JSON and lat/lon formats.
- Converting attestation dictionaries into GeoJSON Features, prioritizing explicit lat/lon fields.
- Building FeatureCollections from lists of attestations.
- Creating JSON content blocks for API responses, including FeatureCollections.

These utilities will streamline the process of generating mapping-friendly outputs from various data sources.

SethDocherty committed:
Adds response extraction utilities

Adds utilities to extract data and pagination information
from Astral MCP API responses.

Handles various response shapes, including those with nested
data structures, and falls back to common keys for broader
compatibility.

SethDocherty committed:
Adds validation and query building helpers

Adds helper functions for validating query parameters and building query parameter dictionaries.

This ensures that query parameters like limit, offset, and prover adhere to specific format and range constraints, improving data integrity and preventing potential errors.

Also, provides a utility to construct query parameter dictionaries, streamlining the process of adding optional parameters to API requests.

SethDocherty committed:
Adds assistant style guide for MCP Inspector

Provides a detailed style guide for crafting consistent and concise assistant commentary when using the MCP Inspector tool.

This guide outlines formatting conventions, standard sections, and error handling procedures to ensure clarity and actionability in assistant responses. It also includes guidance on selecting relevant fields and generating useful URL links for map and media content.

SethDocherty committed:
Adds Copilot instructions for astral-api-mcp

Introduces a comprehensive instruction set for Copilot to enhance its assistance when used with the astral-api-mcp tool.

The instructions guide Copilot's response style, structuring responses with summaries, key details, and next actions, and emphasizing concise, actionable guidance. It also specifies formatting conventions, URL link structures, error handling, and field selection guidance to ensure consistent and helpful interactions with the MCP Inspector.

SethDocherty committed:
Updates black line length

Increases the line length for the black formatter to 130.

This change aims to improve code readability and reduce the number of lines in the codebase, while still maintaining a consistent code style.

SethDocherty committed:
Adds documentation for new MCP tools

Adds documentation for the newly implemented MCP tools,
including querying location proofs, retrieving a specific location
proof by UID, and fetching the Astral API configuration.

SethDocherty committed:
Adds MCP tools guide

Adds a comprehensive guide for using the MCP tools within the Astral API.

This guide provides documentation, example prompts, and usage
patterns for each of the five available tools, including health
checks, server information, location proof queries, and API
configuration retrieval. It also covers best practices, error
handling, and integration examples to facilitate effective use of
the MCP tools.

SethDocherty committed:
Updates README for agent tools clarity

Improves the README by reorganizing the "Available Tools" section to enhance clarity.

Moves the section to be directly under the "Installation" section and renames it to "Available Agent Tools".

This change improves the user experience by making it easier to find information about available tools.

SethDocherty commented on README.md:
The new MCP agent tools that were created were missing references in the main readme

SethDocherty commented on tests/test_server.py:
Test cases here are lacking and very basic.  It's good practice to consider the potential pitfalls and implications of each upstream/downstream component.

SethDocherty commented on astral_mcp_server/server.py:
After testing the experience using VSCode GitHub Copilot, I added additional functionality to extract geoJSON context from attestations, optimizing the Agents' response.

SethDocherty reviewed (approved):
Thanks for working on this @arrnd. You laid the bricks for a solid foundation, and I was able to get the basic functionality up and running. It wasn't immediately clear what new agent tools were available for me to use or how to utilize them. I did encounter a couple of cases where I wasn't getting any results back or an explanation of why. I was able to better identify the issues after updating some of the test cases.  

While you met the low-hanging MVP, there was ample opportunity for you to go the extra mile and make some solid additions, such as exploring MCP use cases,  the response UI, and starting to build some of the documentation.

After reviewing and expanding on the foundation you laid, I'm gonna go ahead and approve and merge the work
